cmake/FindOpenVDB.cmake

```diff
@@ -24,75 +24,478 @@
+    OpenVDB_PATCH_VERSION "${openvdb_version_str}"
+    )
+  UNSET ( openvdb_version_str )
+  UNSET ( OPENVDB_VERSION_FILE )
+  SET ( OpenVDB_VERSION ${OpenVDB_MAJOR_VERSION}.${OpenVDB_MINOR_VERSION}.${OpenVDB_PATCH_VERSION} )
+ENDIF ()
+
+# ------------------------------------------------------------------------
+#  Search for OPENVDB lib DIR
+# ------------------------------------------------------------------------
+
+SET ( _OPENVDB_LIBRARYDIR_SEARCH_DIRS "" )
+
+# Append to _OPENVDB_LIBRARYDIR_SEARCH_DIRS in priority order
+
+LIST ( APPEND _OPENVDB_LIBRARYDIR_SEARCH_DIRS
+  ${OPENVDB_LIBRARYDIR}
+  ${_OPENVDB_ROOT_SEARCH_DIR}
+  ${_OPENVDB_INSTALLED_LIB_DIR}
+  ${PC_OpenVDB_LIBRARY_DIRS}
+  ${SYSTEM_LIBRARY_PATHS}
   )
 
-IF( OpenVDB_FOUND )
-  SET( OpenVDB_INCLUDE_DIR ${OPENVDB_LOCATION}/include
-    CACHE PATH "OpenVDB include directory")
+# Build suffix directories
+
+SET ( OPENVDB_PATH_SUFFIXES
+  lib64
+  lib
+)
+
+# Static library setup
+IF ( UNIX AND OPENVDB_USE_STATIC_LIBS )
+  SET ( _OPENVDB_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES} )
+  SET ( CMAKE_FIND_LIBRARY_SUFFIXES ".a" )
+ENDIF ()
 
-  SET( OpenVDB_LIBRARY_DIR ${OPENVDB_LOCATION}/lib
-    CACHE PATH "OpenVDB library directory" )
+SET ( OpenVDB_LIB_COMPONENTS "" )
 
-  FIND_LIBRARY( OpenVDB_OPENVDB_LIBRARY openvdb
-    PATHS ${OpenVDB_LIBRARY_DIR}
+FOREACH ( COMPONENT ${OpenVDB_FIND_COMPONENTS} )
+  SET ( LIB_NAME ${COMPONENT} )
+  FIND_LIBRARY ( OpenVDB_${COMPONENT}_LIBRARY ${LIB_NAME}
     NO_DEFAULT_PATH
-    NO_SYSTEM_ENVIRONMENT_PATH
+    PATHS ${_OPENVDB_LIBRARYDIR_SEARCH_DIRS}
+    PATH_SUFFIXES ${OPENVDB_PATH_SUFFIXES}
+    )
+  LIST ( APPEND OpenVDB_LIB_COMPONENTS ${OpenVDB_${COMPONENT}_LIBRARY} )
+
+  IF ( OpenVDB_${COMPONENT}_LIBRARY )
+    SET ( OpenVDB_${COMPONENT}_FOUND TRUE )
+  ELSE ()
+    SET ( OpenVDB_${COMPONENT}_FOUND FALSE )
+  ENDIF ()
+ENDFOREACH ()
+
+IF ( UNIX AND OPENVDB_USE_STATIC_LIBS )
+  SET ( CMAKE_FIND_LIBRARY_SUFFIXES ${_OPENVDB_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES} )
+  UNSET ( _OPENVDB_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES )
+ENDIF ()
+
+# ------------------------------------------------------------------------
+#  Cache and set OPENVDB_FOUND
+# ------------------------------------------------------------------------
+
+INCLUDE ( FindPackageHandleStandardArgs )
+FIND_PACKAGE_HANDLE_STANDARD_ARGS ( OpenVDB
+  FOUND_VAR OpenVDB_FOUND
+  REQUIRED_VARS
+    OpenVDB_INCLUDE_DIR
+    OpenVDB_LIB_COMPONENTS
+  VERSION_VAR OpenVDB_VERSION
+  HANDLE_COMPONENTS
+)
+
+IF ( NOT OpenVDB_FOUND )
+  IF ( OpenVDB_FIND_REQUIRED )
+    MESSAGE ( FATAL_ERROR "Unable to find OpenVDB" )
+  ENDIF ()
+  RETURN ()
+ENDIF ()
+
+# ------------------------------------------------------------------------
+#  Determine ABI number
+# ------------------------------------------------------------------------
+
+# Set the ABI number the library was built against. Uses vdb_print
+
+IF ( EXISTS ${_OPENVDB_INSTALLED_BIN_DIR} )
```

Comment:
This variable seems to use a different directory to the one that vdb_print is installed into? The one I've installed to (/usr/local/bin/) is on the $PATH but isn't picked up here?